packages/yaml/src/yaml-document.ts

```diff
@@ -228,6 +228,36 @@ export class YamlDocument {
         return this;
     }
 
+    /**
+     * Deletes a node in a sequence in the document.
+     *
+     * @param path - hierarchical path where the node will be deleted
+     * @param {string} path.path - the path object's path
+     * @param {Object} path.matcher - key/value pair identifying the object
+     * @param {string} path.matcher.key - the key
+     * @param {string} path.matcher.value - the value
+     * @returns {YamlDocument} the YamlDocument instance
+     * @memberof YamlDocument
+     */
+    deleteAt({ path, matcher }: { path: string; matcher: { key: string; value: string } }): YamlDocument {
+        const pathArray = this.toPathArray(path);
+        const seq = this.document.getIn(pathArray) as YAMLSeq<yaml.Node>;
+        if (!seq) {
+            throw new Error(t('error.seqDoesNotExist', { path }));
+        }
```

Comment:
We would need to the case when the node exists but is not a `YAMLSeq`. Otherwise the code after will fail.

Reply:
8c7e5b3